### PR TITLE
[Streaming]Add RefreshChannelInfo to support flow-control

### PR DESCRIPTION
**Pull request by MissiontoMars**
## Why are these changes needed?
`RefreshChannelInfo ` interface is added in `ProducerChannel` and `ConsumerChannel` to support flow-control feature in the following PR.

On producer side, `RefreshChannelInfo` gets the consumed offset of the peer consumer. On consumer side, `RefreshChannelInfo` gets the last seq-id received by StreamingQueue.

In addition, `channel_info ` is renamed to `channel_info _`.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by MissiontoMars** [comment]
@ashione @jovany-wang @chaokunyang

File: streaming/src/channel.h [hunk 1]
```diff
@@ -165,6 +171,7 @@ class MockConsumer : public ConsumerChannel {
                                           uint64_t checkpoint_offset) override {
     return StreamingStatus::OK;
   }
+  StreamingStatus RefreshChannelInfo() {}
```
Comment (ashione) [inline streaming/src/channel.h]: Minconsumed id can be fetched from static map?
Should we implement it here for unittest?

Comment (MissiontoMars) [reply streaming/src/channel.h]: > Minconsumed id can be fetched from static map?
> Should we implement it here for unittest?

If we want to mock and test `RefreshChannelInfo` in MockQueue, ChannelInfo should be added in MockQueue.

**Review by ashione — APPROVED** [review]

File: streaming/src/channel.h [hunk 2]
```diff
@@ -147,6 +151,8 @@ class MockProducer : public ProducerChannel {
     return StreamingStatus::OK;
   }
 
+  StreamingStatus RefreshChannelInfo() {}
```
Comment (chaokunyang) [inline streaming/src/channel.h]: add override

File: streaming/src/channel.h (hunk 1 above)
Comment (chaokunyang) [inline streaming/src/channel.h]: add `override`

Comment (MissiontoMars) [reply streaming/src/channel.h]: > add `override`

done.

File: streaming/src/channel.h (hunk 2 above)
Comment (MissiontoMars) [reply streaming/src/channel.h]: > add override
done

**Review by chaokunyang — APPROVED** [review]

**Pull request merged**